third_party/nvidia_sdk_download/parse_redist.py

```diff
@@ -0,0 +1,302 @@
+#!/usr/bin/env python3
+# Copyright 2021, NVIDIA Corporation
+# SPDX-License-Identifier: MIT
+"""
+Sample parser for redistrib JSON manifests
+1. Downloads each archive
+2. Validates SHA256 checksums
+3. Extracts archives
+4. Flattens into a collapsed directory structure
+
+Forked from https://github.com/NVIDIA/build-system-archive-import-examples/commit/355e25cca11725e88984443a6a343dffeb43308a
+and patched:
+  - avoid a dependency on the non-standard requests package (see
+the http_get helper) by using urllib directly
+  - explicit error handling on hash mismatch
+  - always download, even if files exist
+  - remove dependence on deprecated distutils copy_tree in favor of shutil
```

Comment:
It would be helpful to structure this PR as two commits (and then merge without squashing): one that adds a pristine copy of this file from that link, then a second commit that applies the listed changes.

Reply:
I can land it that way. Please proceed with reviewing as if done.

Reply:
Two commits is also easier to review (saves a few steps of opening the original file and manually diffing, when GitHub's UI can shows diffs between two commits already)

Reply:
Split.